APP_DEVELOPMENT.md

```diff
@@ -79,3 +104,9 @@ We have the [react-native-svg](https://github.com/software-mansion/react-native-
 - Import it at the top of your component file, e.g. `import StaLogoWide from '@/NativeBase/Assets/Images/Logos/sta-logo-wide.svg'`
 - Use it like you would a normal component, e.g. `<StaLogoWide />`
 - See `interface SvgProps` [in this file](https://github.com/software-mansion/react-native-svg/blob/main/src/elements/Svg.tsx) for common props you can use, and [here for touch events](https://github.com/software-mansion/react-native-svg/blob/main/USAGE.md#touch-events)
+
+## Performance issues
+
+Have noticed occasional performance issues in the app emulator e.g. warnings that `serializablestateinvariantmiddleware` took a long time.  We could change the `getDefaultMiddleware` call in `app/Store/index.ts` to set `immutableCheck` and `serializableCheck` to false like [this suggests](https://stackoverflow.com/a/70288486/15422951) (or you could do this when developing locally if needed) -- but this should be avoided if possible.
```

Comment:
I don't _think_ I've done anything untoward in this new code that's slowing down the app, but did notice it being slow at some points - especially when changing dark mode.  But even when I stripped back the code to only using NativeBase's `toggleColorMode` method it could still be slow.

If you see anything daft I've done anywhere in the code that's slowing things down, please do say!

Reply:
Interesting - I thought the slow down was coming from Main.tsx where the Entypo icons are being passed in:

_"Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types)."_

Reply:
Aha, interesting!  Good shout @DougieWougie -- I had introduced something similar in `ApplicationNavigator` which could explain the performance issues.  I've just pushed an update to fix that now, and anecdotally performance seems better in my emulator.

Reply:
Hi @joefhall and @DougieWougie . Does the dark mode suppose to work only in Containers and Profile only at the moment? When I run the Vol App that was the result. Any suggestions?

Reply:
> Hi @joefhall and @DougieWougie . Does the dark mode suppose to work only in Containers and Profile only at the moment? When I run the Vol App that was the result. Any suggestions?

It doesn't apply dark mode in some screens that aren't configured to support it yet. Profile does, so see how that responds?

Reply:
Thanks @Dougie.